File: 136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        int n=nums.length;
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        for (int i=0;i<n;i++) 
        {
            min = Math.min(min, nums[i]);
            max = Math.max(max, nums[i]);
        }

        int[] freq = new int[max-min+1];
        for (int i=0;i<n;i++) 
        {
            freq[nums[i]-min]++;
        }
        for (int i=0;i<n;i++) 
        {
            if (freq[nums[i]-min]==1) {
                return nums[i];
            }
        }

        return -1;
    }
}

File: 136-single-number/single-number.c
void solve(int* nums,int numsSize)
{
    for(int i=0;i<numsSize-1;i++)
    {
        for(int j=0;j<numsSize-i-1;j++)
        {
            if(nums[j]>nums[j+1])
            {
                int temp=nums[j];
                nums[j]=nums[j+1];
                nums[j+1]=temp;
            }
        }
    }
}
int singleNumber(int* nums, int numsSize) {
    solve(nums,numsSize);
    for(int i=1;i<numsSize;i+=2)
    {
        if(nums[i]!=nums[i-1])
        {
            return nums[i-1];
        }
    }
    return nums[numsSize-1];
}
